runtime/onert/core/src/util/shapeinf/Range.cc

```diff
@@ -0,0 +1,101 @@
+
+  out_shape.dim(0) =
+      (std::is_integral<T>::value
+           ? ((std::abs(*start_val - *limit_val) + std::abs(*delta_val) - 1) / std::abs(*delta_val))
+           : std::ceil(std::abs((*start_val - *limit_val) / *delta_val)));
+  return out_shape;
+}
+
+// StaticInferer at compilation time
+void StaticInferer::visit(const ir::operation::Range &op)
+{
+  const auto start_idx{op.getInputs().at(ir::operation::Range::Input::START)};
+  const auto limit_idx{op.getInputs().at(ir::operation::Range::Input::LIMIT)};
+  const auto delta_idx{op.getInputs().at(ir::operation::Range::Input::DELTA)};
+  const auto &start_op = _operands.at(start_idx);
+  const auto &limit_op = _operands.at(limit_idx);
+  const auto &delta_op = _operands.at(delta_idx);
+
+  // get mutable output operand
+  const auto output_idx = op.getOutputs().at(0);
+  ir::Operand &output = _operands.at(output_idx);
+  // if any input is dynamic, output also becomes dynamic
+  if (start_op.info().isDynamic() || limit_op.info().isDynamic() || delta_op.info().isDynamic())
+  {
+    output.info().setDynamic();
+    return;
+  }
+}
+
+// DynamicInferer at execution time
+void DynamicInferer::visit(const ir::operation::Range &op)
+{
+  // check if output is not dynamic
+  auto output_ind = op.getOutputs().at(0);
+  auto output = _tensor_registry->getITensor(output_ind);
+
+  // from op, access the buffer of second input to read new shape
+  auto start_idx = op.getInputs().at(ir::operation::Range::Input::START);
+  auto start_tensor = _tensor_registry->getITensor(start_idx);
+  auto start_shape = getShape(start_tensor.get());
+
+  auto limit_idx = op.getInputs().at(ir::operation::Range::Input::LIMIT);
+  auto limit_tensor = _tensor_registry->getITensor(limit_idx);
+  auto limit_shape = getShape(limit_tensor.get());
+
+  auto delta_idx = op.getInputs().at(ir::operation::Range::Input::DELTA);
+  auto delta_tensor = _tensor_registry->getITensor(delta_idx);
+  auto delta_shape = getShape(delta_tensor.get());
+
+  if (!start_tensor->is_dynamic() && !limit_tensor->is_dynamic() && !delta_tensor->is_dynamic())
+    return;
+
+  if (output->data_type() == ir::DataType::FLOAT32)
+  {
+    ir::Shape new_shape =
+        inferRangeShape<float *>(reinterpret_cast<float *>(start_tensor->buffer()),
+                                 reinterpret_cast<float *>(limit_tensor->buffer()),
+                                 reinterpret_cast<float *>(delta_tensor->buffer()));
+    setShape(output.get(), new_shape);
+    _dynamic_tensor_manager->applyShape(output_ind, new_shape);
+  }
+  else if (output->data_type() == ir::DataType::INT32)
+  {
+    ir::Shape new_shape =
+        inferRangeShape<int32_t *>(reinterpret_cast<int32_t *>(start_tensor->buffer()),
+                                   reinterpret_cast<int32_t *>(limit_tensor->buffer()),
+                                   reinterpret_cast<int32_t *>(delta_tensor->buffer()));
+    setShape(output.get(), new_shape);
+
```

Comment:
I think setShape() and applyShape() are redundant.
I think we can invoke both function outside of if ~ else if statement : )

Reply:
WIll remove the redundant SetShape. As `new_shape` is declared and initialized inside the respective blocks, `applyShape` is called immediately after the initialization.

Reply:
How about declaring new_shape outside?

Reply:
might work, it does not have copy constructor, but has an `=` overloaded. If that works, then I will upload the change. Thanks.